cpp/thirdparty/versions.txt

```diff
@@ -121,7 +121,7 @@ DEPENDENCIES=(
   "ARROW_GLOG_URL glog-${ARROW_GLOG_BUILD_VERSION}.tar.gz https://github.com/google/glog/archive/${ARROW_GLOG_BUILD_VERSION}.tar.gz"
   "ARROW_GOOGLE_CLOUD_CPP_URL google-cloud-cpp-${ARROW_GOOGLE_CLOUD_CPP_BUILD_VERSION}.tar.gz https://github.com/googleapis/google-cloud-cpp/archive/refs/tags/${ARROW_GOOGLE_CLOUD_CPP_BUILD_VERSION}.tar.gz"
   "ARROW_GRPC_URL grpc-${ARROW_GRPC_BUILD_VERSION}.tar.gz https://github.com/grpc/grpc/archive/${ARROW_GRPC_BUILD_VERSION}.tar.gz"
-  "ARROW_GTEST_URL gtest-${ARROW_GTEST_BUILD_VERSION}.tar.gz https://github.com/google/googletest/archive/release-${ARROW_GTEST_BUILD_VERSION}.tar.gz"
+  "ARROW_GTEST_URL {ARROW_GTEST_BUILD_VERSION}.tar.gz https://github.com/google/googletest/archive/${ARROW_GTEST_BUILD_VERSION}.tar.gz"
```

Comment:
Could you add `googletest-` prefix for easy to know package name:

```console
$ curl -L -v https://github.com/google/googletest/archive/af29db7ec28d6df1c7f0f745186884091e602e07.tar.gz 2>&1 | grep filename= 
< content-disposition: attachment; filename=googletest-af29db7ec28d6df1c7f0f745186884091e602e07.tar.gz
```

Reply:
Just adding `googletest-` is sufficient right? I've added that back in